### PR TITLE
support spatial_scale >= 1.0 in roi_average_align_2d.py

### PR DESCRIPTION
**Pull request by knorth55**
currently, when `spatial_scale > 1.0` is given, the function hangs.

### CODE REVIEW

**Comment by knorth55** [comment]
@Hakuyume told me to refactor CUDA kernel and other loop.

**Comment by knorth55** [comment]
@Hakuyume I updated to reduce loop trials. Can you review again?

File: chainer/functions/pooling/roi_average_align_2d.py [hunk 1]
```diff
@@ -30,92 +30,66 @@ def _pair(x):
     return x, x
 
 
-def _get_bilinear_interp_params(y, x, height, width):
-    if y < -1 or y > height or x < -1 or x > width:
+def _get_bounds(p, limit):
+    if p < -1 or p > limit:
```
Comment (Hakuyume) [inline chainer/functions/pooling/roi_average_align_2d.py]: `p <= 0` ?

**Comment by Hakuyume** [comment]
Thanks. I'll check.

File: chainer/functions/pooling/roi_max_align_2d.py [hunk 2]
```diff
@@ -127,16 +128,19 @@ def forward_cpu(self, inputs):
                 y = roi_start_h + ph * bin_size_h + \
                     (iy + .5) * bin_size_h / roi_bin_grid_h
                 ix = 0
+                y, y_low, y_high = _get_bounds(y, height)
+                if y is None or y_low is None or y_high is None:
+                    continue
                 while ix < roi_bin_grid_w:
```
Comment (Hakuyume) [inline chainer/functions/pooling/roi_max_align_2d.py]: `for`

Comment (knorth55) [reply chainer/functions/pooling/roi_max_align_2d.py]: this PR is for `roi_average_align_2d`.
see https:/github.com/chainer/chainer/pull/5635 for `roi_max_align_2d.py`.

**Comment by Hakuyume** [comment]
jenkins, test this please.

**Pull request merged**